### PR TITLE
Use the logging agent's node name as the metadata agent URL.

### PR DESCRIPTION
**Pull request by bmoyles0117**
The Stackdriver Logging agent should use the node's hostname when it constructs the Stackdriver Metadata Agent's URL, currently, it's using the GKE Master's hostname, which is a bug.

**Release note:**
```release-note
[fluentd-gcp addon] Use the logging agent's node name as the metadata agent URL.
```

### CODE REVIEW

**Comment by bmoyles0117** [comment]
@kawych @piosz @rphillips can someone please review and add the ok-to-test label?

File: cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml [hunk 1]
```diff
@@ -38,13 +38,13 @@ spec:
...
         env:
-        - name: STACKDRIVER_METADATA_AGENT_URL
-          value: {{ stackdriver_metadata_agent_url }}
         - name: NODE_NAME
           valueFrom:
             fieldRef:
               apiVersion: v1
               fieldPath: spec.nodeName
+        - name: STACKDRIVER_METADATA_AGENT_URL
+          value: http://$(NODE_NAME):8799
```
Comment (kawych) [inline cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: $(NODE_NAME) is not visible here. Please generate the url in configure-helper.sh and substitute it with sed as it is done now. Please make sure to test that it works as intended before submitting.

**Review by kawych — CHANGES_REQUESTED** [review]

**Comment by kawych** [comment]
/ok-to-test
Are you going to cherry-pick it to 1.10 branch too?

File: cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml (hunk 1 above)
Comment (bmoyles0117) [reply cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Node name is visible here via Kubernetes bash variable substitution. This is how we're doing it in our manual installation instructions, without the use of bash sed replacing. You can see the configuration as we apply it verbatim here in our public config: https://storage.googleapis.com/stackdriver-gke-beta/configs/logging-agent.yaml 

I can try to find the official documentation for this feature, but it has been tested and validated by our team.

**Comment by bmoyles0117** [comment]
Yes, we will cherry pick this into 1.10 if we can get it approved in time.

**Pull request merged**